In [1]:
import pandas as pd

ModuleNotFoundError: No module named 'pandas'

In [6]:
Customers = pd.read_csv("../data/Customers.csv")
Orders = pd.read_csv("../data/Orders.csv")
Order_Details = pd.read_csv("../data/Order_Details.csv")
Products = pd.read_csv("../data/Products.csv")
Targets = pd.read_csv("../data/Targets.csv")

Cheak Column Names

In [21]:
print("Customers Columns:", Customers.columns.tolist())
print("Products Columns:", Products.columns.tolist())
print("Orders Columns:", Orders.columns.tolist())
print("Order Details Columns:", Order_Details.columns.tolist())
print("Targets Columns:", Targets.columns.tolist())

Customers Columns: ['Customer_ID', 'Customer_Name', 'Gender', 'Age', 'City', 'State', 'Region']
Products Columns: ['Product_ID', 'Product_Name', 'Category', 'Sub_Category', 'Brand', 'Cost_Price', 'Selling_Price']
Orders Columns: ['Order_ID', 'Order_Date', 'Customer_ID', 'Payment_Method', 'Order_Status', 'Year', 'Month_Name', 'Quarter', 'Day', 'Weekday', 'Month', 'Region']
Order Details Columns: ['Order_ID', 'Product_ID', 'Quantity', 'Discount', 'Selling_Price', 'Cost_Price', 'Gross_Sales', 'Discount_Amount', 'Net_Sales', 'Total_Cost', 'Profit', 'Profit_Margin', 'Customer_ID', 'Order_Date', 'Month', 'Month_start', 'Region', 'Category']
Targets Columns: ['Month', 'Region', 'Sales_Target', 'Profit_Target']



Data Understanding


In [11]:
print("CUSTOMERS INFO")
Customers.info()


CUSTOMERS INFO
<class 'pandas.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 7 columns):
 #   Column         Non-Null Count  Dtype
---  ------         --------------  -----
 0   Customer_ID    10000 non-null  str  
 1   Customer_Name  10000 non-null  str  
 2   Gender         10000 non-null  str  
 3   Age            10000 non-null  int64
 4   City           10000 non-null  str  
 5   State          10000 non-null  str  
 6   Region         10000 non-null  str  
dtypes: int64(1), str(6)
memory usage: 547.0 KB


In [10]:
print("Orders")
Orders.info()

Orders
<class 'pandas.DataFrame'>
RangeIndex: 400000 entries, 0 to 399999
Data columns (total 12 columns):
 #   Column          Non-Null Count   Dtype
---  ------          --------------   -----
 0   Order_ID        400000 non-null  str  
 1   Order_Date      400000 non-null  str  
 2   Customer_ID     400000 non-null  str  
 3   Payment_Method  400000 non-null  str  
 4   Order_Status    400000 non-null  str  
 5   Year            400000 non-null  int64
 6   Month_Name      400000 non-null  int64
 7   Quarter         400000 non-null  str  
 8   Day             400000 non-null  int64
 9   Weekday         400000 non-null  str  
 10  Month           400000 non-null  str  
 11  Region          400000 non-null  str  
dtypes: int64(3), str(9)
memory usage: 36.6 MB


In [13]:

print("Order_Details")
Order_Details.info()

Order_Details
<class 'pandas.DataFrame'>
RangeIndex: 1019516 entries, 0 to 1019515
Data columns (total 18 columns):
 #   Column           Non-Null Count    Dtype  
---  ------           --------------    -----  
 0   Order_ID         1019516 non-null  str    
 1   Product_ID       1019516 non-null  str    
 2   Quantity         1019516 non-null  int64  
 3   Discount         1019516 non-null  float64
 4   Selling_Price    1019516 non-null  int64  
 5   Cost_Price       1019516 non-null  int64  
 6   Gross_Sales      1019516 non-null  int64  
 7   Discount_Amount  1019516 non-null  float64
 8   Net_Sales        1019516 non-null  float64
 9   Total_Cost       1019516 non-null  int64  
 10  Profit           1019516 non-null  float64
 11  Profit_Margin    1019516 non-null  float64
 12  Customer_ID      1019516 non-null  str    
 13  Order_Date       1019516 non-null  str    
 14  Month            1019516 non-null  str    
 15  Month_start      1019516 non-null  str    
 16  Region         

In [12]:
print("PRODUCT INFO")
Products.info()

PRODUCT INFO
<class 'pandas.DataFrame'>
RangeIndex: 1392 entries, 0 to 1391
Data columns (total 7 columns):
 #   Column         Non-Null Count  Dtype
---  ------         --------------  -----
 0   Product_ID     1392 non-null   str  
 1   Product_Name   1392 non-null   str  
 2   Category       1392 non-null   str  
 3   Sub_Category   1392 non-null   str  
 4   Brand          1392 non-null   str  
 5   Cost_Price     1392 non-null   int64
 6   Selling_Price  1392 non-null   int64
dtypes: int64(2), str(5)
memory usage: 76.3 KB


In [14]:
print("Targets")
Targets.info()

Targets
<class 'pandas.DataFrame'>
RangeIndex: 240 entries, 0 to 239
Data columns (total 4 columns):
 #   Column         Non-Null Count  Dtype
---  ------         --------------  -----
 0   Month          240 non-null    str  
 1   Region         240 non-null    str  
 2   Sales_Target   240 non-null    int64
 3   Profit_Target  240 non-null    int64
dtypes: int64(2), str(2)
memory usage: 7.6 KB


Check Missing Values

In [15]:
print("CUSTOMERS")
display(Customers.isnull().sum())

print("PRODUCTS")
display(Products.isnull().sum())

print("ORDERS")
display(Orders.isnull().sum())

print("ORDER DETAILS")
display(Order_Details.isnull().sum())

print("TARGETS")
display(Targets.isnull().sum())

CUSTOMERS


Customer_ID      0
Customer_Name    0
Gender           0
Age              0
City             0
State            0
Region           0
dtype: int64

PRODUCTS


Product_ID       0
Product_Name     0
Category         0
Sub_Category     0
Brand            0
Cost_Price       0
Selling_Price    0
dtype: int64

ORDERS


Order_ID          0
Order_Date        0
Customer_ID       0
Payment_Method    0
Order_Status      0
Year              0
Month_Name        0
Quarter           0
Day               0
Weekday           0
Month             0
Region            0
dtype: int64

ORDER DETAILS


Order_ID           0
Product_ID         0
Quantity           0
Discount           0
Selling_Price      0
Cost_Price         0
Gross_Sales        0
Discount_Amount    0
Net_Sales          0
Total_Cost         0
Profit             0
Profit_Margin      0
Customer_ID        0
Order_Date         0
Month              0
Month_start        0
Region             0
Category           0
dtype: int64

TARGETS


Month            0
Region           0
Sales_Target     0
Profit_Target    0
dtype: int64

Duplicate Records

In [16]:
print("Customers Duplicates:", Customers.duplicated().sum())
print("Products Duplicates:",Products.duplicated().sum())
print("Orders Duplicates:",Orders.duplicated().sum())
print("Order Details Duplicates:",Order_Details.duplicated().sum() )
print("Targets Duplicates:",Targets.duplicated().sum())

Customers Duplicates: 0
Products Duplicates: 0
Orders Duplicates: 0
Order Details Duplicates: 0
Targets Duplicates: 0


Statistical Summary

In [17]:
# Customers Dataset
Customers.describe()

,Age
count,10000.000000
mean,29.279400
std,7.379131
min,18.000000
25%,24.000000
50%,29.000000
75%,34.000000
max,59.000000


In [18]:
# Products Dataset
Products.describe()

,Cost_Price,Selling_Price
count,1392.000000,1392.000000
mean,215.851293,285.369253
std,241.038023,323.031850
min,5.000000,10.000000
25%,60.000000,80.000000
50%,130.000000,170.000000
75%,271.250000,357.750000
max,1770.000000,2403.000000


In [19]:
# Order Details Dataset
Order_Details.describe()

,Quantity,Discount,Selling_Price,Cost_Price,Gross_Sales,Discount_Amount,Net_Sales,Total_Cost,Profit,Profit_Margin
count,1.019516e+06,1.019516e+06,1.019516e+06,1.019516e+06,1.019516e+06,1.019516e+06,1.019516e+06,1.019516e+06,1.019516e+06,1.019516e+06
mean,1.703572e+00,6.500153e-02,2.484393e+02,1.928705e+02,4.005277e+02,2.603102e+01,3.744967e+02,3.117879e+02,6.270878e+01,1.713145e-01
std,9.484181e-01,6.643612e-02,2.732606e+02,2.131841e+02,5.088521e+02,5.422104e+01,4.776463e+02,4.021610e+02,1.035539e+02,1.115975e-01
min,1.000000e+00,0.000000e+00,1.000000e+01,5.000000e+00,1.000000e+01,0.000000e+00,7.500000e+00,5.000000e+00,-1.142000e+03,-3.492063e-01
25%,1.000000e+00,0.000000e+00,6.500000e+01,5.000000e+01,1.000000e+02,0.000000e+00,9.000000e+01,7.500000e+01,1.000000e+01,1.000000e-01
50%,1.000000e+00,5.000000e-02,1.500000e+02,1.200000e+02,2.220000e+02,7.000000e+00,2.090000e+02,1.750000e+02,2.950000e+01,1.700000e-01
75%,2.000000e+00,1.000000e-01,3.200000e+02,2.350000e+02,5.100000e+02,2.775000e+01,4.750000e+02,4.000000e+02,7.500000e+01,2.500000e-01
max,5.000000e+00,3.000000e-01,2.403000e+03,1.770000e+03,8.640000e+03,1.620000e+03,8.640000e+03,6.760000e+03,2.664000e+03,5.000000e-01


DATA SUMMARY

In [20]:
data_summary = pd.DataFrame({
    "Dataset": [
        "Customers",
        "Products",
        "Orders",
        "Order_Details",
        "Targets"
    ],
    "Rows": [
        Customers.shape[0],
        Products.shape[0],
        Orders.shape[0],
        Order_Details.shape[0],
        Targets.shape[0]
    ],
    "Columns": [
        Customers.shape[1],
        Products.shape[1],
        Orders.shape[1],
        Order_Details.shape[1],
        Targets.shape[1]
    ],
    "Duplicates": [
        Customers.duplicated().sum(),
        Products.duplicated().sum(),
        Orders.duplicated().sum(),
        Order_Details.duplicated().sum(),
        Targets.duplicated().sum()
    ]
})

display(data_summary)

,Dataset,Rows,Columns,Duplicates
0,Customers,10000,7,0
1,Products,1392,7,0
2,Orders,400000,12,0
3,Order_Details,1019516,18,0
4,Targets,240,4,0


Initial Observations

1. The Customers dataset contains 10,000 records and 7 columns.

2. The Products dataset contains 1,392 records and 7 columns.

3. The Orders dataset contains 400,000 records and 12 columns.

4. The Order_Details dataset contains 1,019,516 records and 18 columns.

5. The Targets dataset contains 240 records and 4 columns.

6. No exact duplicate rows were found in any of the five datasets.

Conclusion

The five datasets were examined using Pandas.

The analysis included:
- Dataset dimensions
- Column names
- Data types
- Missing value checks
- Duplicate record checks
- Statistical summaries

No exact duplicate rows were found in the datasets.
